### ISSUE TITLE
Mention multi-collinearity problem with categorical variables encoding.

### ISSUE THREAD

**Issue by nkthiebaut** [christophM/interpretable-ml-book#143]
First thanks a lot for writing this book, it's a gold mine of knowledge and an exegesis masterpiece!

I've noticed that the dummy encoding scheme with one less variable  (sometime called "one-hot-encoding", sometimes called "dummy encoding" although it is slightly different from yours) is not mentioned in the section 4.1.5 ("Encoding of Categorical Features"), and all the methods proposed suffer from multi-collinearity because they encode variables with as many columns as the initial number of categories (see, e.g. https://towardsdatascience.com/one-hot-encoding-multicollinearity-and-the-dummy-variable-trap-b5840be3c41a or the answer to this Stack Overflow question: https://stats.stackexchange.com/questions/224051/one-hot-vs-dummy-encoding-in-scikit-learn). In short, when encoding categorical variables with $m$ categories with $m$ columns then every such column is a linear combination of the other $m-1$ columns, which is bad for the loss function minimization because it creates an infinite number of equivalent solutions, and is bad for weights interpretation too.

Shouldn't this paragraph mention the multi-collinearity problem and dummy encoding with one less column? 

If you agree I can make a PR with additional comments.

**Comment by christophM** [comment]
thanks!
